### DISCUSSION TITLE
Concatenating files with "get-content | add-content" failes

### DISCUSSION THREAD

**Discussion by User843212** [PowerShell/PowerShell#15183] [Q&A]
Windows 10 x64 - PowerShell 7.1.3 x64

I have downloaded video files "segment1.ts ... segmentN.ts" and want to concatenate these files to one single "Video.ts" 

--- bad  code:   - filenames are stored in array "$Segments" -

$Segments | ForEach-Object { Get-Content $_ -Raw | Add-Content "Video.ts"  }

This code does not work in PowerShell 7  ...  but works perfectly in PowerShell 5
It seems that the file "Video.ts" **is twice as big** when created in PS 7 as opposed to PS 5.

--- good code:

$Segments | ForEach-Object { Get-Content $_ -AsByteStream -ReadCount 0 | Add-Content "Video.ts" -AsByteStream  }

This version works but very slow!

Is it  possible that "PS7 x64" has a problem with "int32" to "int64" conversion in "get/add-content"?

**Comment by AltitudeApps** [comment]
Other than using native .NET file operations, (because `Get-Content` isn't the best fit for what you are doing.), you might try removing the `ForEach-Object`.  

This is because `Get-Content` already accepts input from the pipeline, so there is no need to adapt it for pipeline use via `ForEach-Object`.

```pwsh
$thisMany = 1MB
$Segments | Get-Content -AsByteStream -ReadCount $thisMany | Add-Content "Video.ts" -AsByteStream 
```

Also, try a different values for `-ReadCount`, because the advice given on the docs page for `Get-Content` isn't really anticipating it being used to work with files of such a large size.   With `-Readcount 0`, you are asking it to read the __ENTIRE__ file into memory, before sending that data into the pipeline, where `Add-Content` will receive it.
